reference/src/layout/arrays.md

```diff
@@ -0,0 +1,76 @@
+# Layout of Rust array types
+
+Array types, `[T; N]`, store `N` values of type `T` with a constant _stride_.
+Here, _stride_ is the distance between each pair of consecutive values within
+the array.
+
+The _offset_ of the first array element is `0`, that is, a pointer to the array
+and a pointer to its first element both point to the same memory address.
+
+The _alignment_ of array types is greater or equal to the alignment of its
+element type. If the element type is `repr(C)` the layout of the array is
+guaranteed to be the same as the layout of a C array with the same element type.
+
+> **Note**: the type of array arguments in C function signatures, e.g., `void
+> foo(T x[N])`, decays to a pointer. That is, these functions do not take arrays
+> as an arguments, they take a pointer to the first element of the array
+> instead. Array types are therefore _improper C types_ (not C FFI safe) in Rust
+> foreign function declarations, e.g., `extern { fn foo(x: [T; N]) -> [U; M];
+> }`. Pointers to arrays are fine: `extern { fn foo(x: *const [T; N]) -> *const
+> [U; M]; }`.
+
+The _stride_ of the array is the distance between a pair of consecutive values
+within the array, and it is constant for all element pairs. It is computed as
+the _size_ of the element type rounded up to the next multiple of the
+_alignment_ of the element type.
+
+When the element _size_ is a multiple of the element's _alignment_, then `stride
```

Comment:
What about putting this and the next paragraph into a subsection like "Special case: stride == size"?

Reply:
Sounds good !